cpp/src/fil/infer.cu

```diff
@@ -182,7 +182,7 @@ struct tree_aggregator_t {
     acc = BlockReduce<NITEMS>(*(TempStorage*)tmp_storage).Sum(acc);
     if (threadIdx.x == 0) {
       for (int i = 0; i < NITEMS; ++i) {
-        int row = blockIdx.x * NITEMS + i;
+        int row = i;
```

Comment:
I didn't realize that changing the meaning of the `out` parameter would have this effect.

In this case, it's better to have a single variable:
```
#pragma unroll
for (int row = 0; row < NITEMS; ++row) {
  if (row < num_rows) out[row * output_stride] = acc[row];
}
```